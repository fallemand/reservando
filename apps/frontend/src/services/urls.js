const Urls = {
  FETCH_PRODUCT_INFO: "/fetch",
  UPDATE_PRODUCT_INFO: "/edit",
  UPLOAD_PHOTO: "/tour_photo_upload",
  DELETE_PHOTO: "/tour_photo_delete",
  UPDATE_PHOTOS_ORDER: "/tour_photo_update_order",
  CREATE_PRICING: "/tour_pricing_create",
  DELETE_PRICING: "/tour_pricing_delete",
  FETCH_ADDONS: "/addons_fetch",
  TAGS_AUTOCOMPLETE: "/tags_fetch",
  TAGS_RECOMMENDATIONS: "/tags_recommendations",
  KEYWORDS_SUGGESTIONS: "/keywords_suggestion",
  CREATE_AVAILABILITY: "/tour_availability_block_create",
  DELETE_AVAILABILITY: "/tour_availability_block_delete",
  SUBMIT_PRODUCT: "/tour_activate",
  FETCH_AVAILABLE_RESERVATION_SYSTEMS: "/ticket_source_options",
};

export default Urls;
