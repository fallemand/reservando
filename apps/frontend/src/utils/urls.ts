const urls = {
  home: "/",
  login: "/login",
  termsOfService: "/home",
  privacyPolicy: "/home",
  loginRedirect: (redirectUrl: string): string => {
    // Avoid adding the param infinitely
    if (redirectUrl.includes("redirectUrl")) {
      return urls.login;
    }

    return `/login/#/?redirectUrl=${redirectUrl}`;
  },
};

export default urls;
