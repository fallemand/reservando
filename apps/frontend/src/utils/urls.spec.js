import urls from "./urls";

describe("URL utility", () => {
  it("should have a list of urls", () => {
    expect(urls).toMatchSnapshot();
  });
});
