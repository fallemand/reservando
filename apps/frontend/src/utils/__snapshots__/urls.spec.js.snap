// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URL utility should have a list of urls 1`] = `
Object {
  "home": "/",
  "login": "/login",
  "loginRedirect": [Function],
  "privacyPolicy": "/home",
  "termsOfService": "/home",
}
`;
